scripts/migration/migrate-to-asm

```diff
@@ -1337,13 +1495,90 @@ apply_backup() {
 }
 
 remove_asm() {
-  run istioctl x uninstall --revision "${REVISION_LABEL}" -y
+  run istioctl x uninstall --revision "$(asm_revision_label)" -y
   run kubectl delete ns asm-system
 
   info "****"
   info "Anthos Service Mesh has been uninstalled from the cluster."
 }
 
+should_use_mcp() {
+  if [[ "${NO_MCP}" -eq 1 ]]; then false; return; fi
+  if [[ -f ./configuration_backup/use_mcp ]]; then return; fi
+  if [[ -f ./configuration_backup/no_use_mcp ]]; then false; return; fi
+
+  run_mcp_fit_assessment
+}
+
+is_private_cluster() {
+  gcloud container clusters describe \
+    --region "${CLUSTER_LOCATION}" \
+    --project "${PROJECT_ID}" \
+    "${CLUSTER_NAME}" --format json \
+    | grep -q '"enablePrivateEndpoint": true,'
+}
+
+is_autopilot() {
+  gcloud container clusters describe \
+    --region "${CLUSTER_LOCATION}" \
+    --project "${PROJECT_ID}" \
+    "${CLUSTER_NAME}" --format json \
+    | grep -q 'autopilot'
+}
+
+is_supported_location() {
+  while read -r region; do
+    if [[ "${CLUSTER_LOCATION}" = "${region}"* ]]; then
```

Comment:
is CLUSTER_LOCATION always region? never something like us-central1-c?

Reply:
Nope, I'll change the variable name. Lazy copy/paste.

Reply:
Nevermind, actually it is always region--these are the supported regions for MCP, not anything the user input.